src/libostree/ostree-core.c

```diff
@@ -2014,7 +2014,7 @@ file_header_parse (GVariant         *metadata,
   mode = GUINT32_FROM_BE (mode);
   g_autoptr(GFileInfo) ret_file_info = _ostree_mode_uidgid_to_gfileinfo (mode, uid, gid);
 
-  if (S_ISREG (mode))
+  if (S_ISREG (mode) || S_ISCHR(mode))
```

Comment:
I think we should have a common helper that checks specifically for whiteouts, which needs to take either a `GFileInfo` or a `struct stat`.

We still do want to error out for any character devices that are not whiteouts.